### ISSUE TITLE
FreeBSD (under Freenas jail)

### ISSUE THREAD

**Issue by onurbaser** [CollaboraOnline/richdocumentscode#35]
If I am not wrong, Glibc is not available under freebsd. So is there any way to use an alternative to this System Requirement? Thank you

Nextcloud @ Freenas jail (FreeNAS-11.2-U7)

**Comment by RafalLukawiecki** [comment]
I am also running NC in a FreeNAS/FreeBSD jail. Using the NC web UI the installation of both Collabora CODE and the integration app proceeded without an error, but I am not sure how to configure it further. I see these in the NC Settings:

<img width="995" alt="Screenshot 2020-06-30 at 14 40 19" src="https://user-images.githubusercontent.com/22912705/86133001-a06a0f00-badf-11ea-820a-eb238eaec33e.png">
<img width="1226" alt="Screenshot 2020-06-30 at 14 40 33" src="https://user-images.githubusercontent.com/22912705/86133029-a9f37700-badf-11ea-83ad-98cc7ca01a22.png">
<img width="893" alt="Screenshot 2020-06-30 at 14 40 56" src="https://user-images.githubusercontent.com/22912705/86133068-b7106600-badf-11ea-8b1c-99aa17d238f1.png">

I can also see them in the www/nextcloud/apps folder. I don't know how functional those are, as I am not sure what URLs etc to use to set it up.

**Comment by onurbaser** [comment]
Yes, same here. I can install both of it without an error but when I point it to local server address, it cannot find. I thought the reason is unmet requirement (Glibc).

**Comment by goerbs** [comment]
The same for me

**Comment by cju77** [comment]
Same issue here!

**Comment by diode-ee** [comment]
Came here to join the list of others that are having this issue. Any fix to this?

**Comment by deneb2** [comment]
same here

**Comment by johnjaylward** [comment]
When I try to install the plugin using the FreeNas/TrueNas plugin I get the following issue:

```
App "Collabora Online - Built-in CODE Server" cannot be installed because the following dependencies are not fulfilled: The following architectures are supported: x86_64
```

I don't see a plugin version that is specific to FreeBSD. My server is x86_64, but is not Linux based.

Is there something I can do to get it working?

**Comment by jdakhayman** [comment]
> When I try to install the plugin using the FreeNas/TrueNas plugin I get the following issue:
> 
> > App "Collabora Online - Built-in CODE Server" cannot be installed because the following dependencies are not fulfilled: The following architectures are supported: x86_64
> 
> I don't see a plugin version that is specific to FreeBSD. My server is x86_64, but is not Linux based.
> 
> Is there something I can do to get it working?

I Have Nextcloud install in a jail on stock Freebsd 12.2-RELEASE. Everything works other than this. I get the same error message as johnjaylward

**Comment by jdakhayman** [comment]
> If I am not wrong, Glibc is not available under freebsd. So is there any way to use an alternative to this System Requirement? Thank you
> 
> Nextcloud @ Freenas jail (FreeNAS-11.2-U7)

> > When I try to install the plugin using the FreeNas/TrueNas plugin I get the following issue:
> > > App "Collabora Online - Built-in CODE Server" cannot be installed because the following dependencies are not fulfilled: The following architectures are supported: x86_64
> > 
> > 
> > I don't see a plugin version that is specific to FreeBSD. My server is x86_64, but is not Linux based.
> > Is there something I can do to get it working?
> 
> I Have Nextcloud install in a jail on stock Freebsd 12.2-RELEASE. Everything works other than this. I get the same error message as johnjaylward


This issue has been here for over 6 months with no response. 

Is it possible to actually get a answer either way?

"Yes we are working on it, its more involved then you think."

"No, it'll never happen."

I completely understand that is is oss. But a simple Yes/No from the devs would at least provide some feedback.

jda

**Comment by kendy** [comment]
@jdakhayman You are right - we should have answered, terribly sorry about that :-(

The thing is that richdocumentscode uses AppImage under thee hood; and the executable there is highly Linux-dependent, so it is unlikely it will work unless somebody ports it to FreeBSD.

Very happy to help you with code pointers if you (or somebody here) would like to help there?
